### PR TITLE
Fix UTF bom parsing

### LINKED ISSUE #9926: JabRef V5.9 still corrupts my BIB file.
### JabRef version

5.9 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10 22H2

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. .Install JabRef v5.9 portable
2. Open my bib file, MHF.bib
3. Save the file
4. try to reopen the file
5. you get an error of no records
6. edit the file in text editor.  It is all binary, not text.

This is the same problem as 
#8895.  This has been broken since version 5.6.

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```
compressed version of my original file.

[MHF - Copy.zip](https://github.com/JabRef/jabref/files/11520347/MHF.-.Copy.zip)

</details>
image

### PR DESCRIPTION
**Pull request by Siedlerchr**
line.starts with fails on utf bom chars, because the % sign is actually at position 1 and not on pos 0 in the string.
java reads the bom chars as well.

grafik

Fixes #9926

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
todo add bib file also as test file

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,15 +76,17 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
...
 - We fixed an issue where an exception was thrown for the user after <kbd>Ctrl</kbd>+<kbd>Z</kbd> command. [#9737](https://github.com/JabRef/jabref/issues/9737)
-- We fixed the citation key generation for (`[authors]`, `[authshort]`, `[authorsAlpha]`, `authIniN`, `authEtAl`, `auth.etal`)[https://docs.jabref.org/setup/citationkeypatterns#special-field-markers] to handle `and others` properly. [koppor#626](https://github.com/koppor/jabref/issues/626)
+- We fixed the citation key generation for [`[authors]`, `[authshort]`, `[authorsAlpha]`, `[authIniN]`, `[authEtAl]`, `[auth.etal]`](https://docs.jabref.org/setup/citationkeypatterns#special-field-markers) to handle `and others` properly. [koppor#626](https://github.com/koppor/jabref/issues/626)
 - We fixed the Save/save as file type shows BIBTEX_DB instead of "Bibtex library". [#9372](https://github.com/JabRef/jabref/issues/9372)
 - We fixed the default main file directory for non-English Linux users. [#8010](https://github.com/JabRef/jabref/issues/8010)
+- We fixed an issue when overwriting the owner was disabled. [#9896](https://github.com/JabRef/jabref/pull/9896)
```
Comment (koppor) [inline CHANGELOG.md:82]: Strange that his comes in

**Review by koppor — APPROVED** [review]
Good catch!

File: CHANGELOG.md (hunk 1 above)
Comment (Siedlerchr) [reply CHANGELOG.md:82]: It was a merge thing. Somehow this was deleted in the main.

**Review by calixtus — APPROVED** [review]

**Comment by calixtus** [comment]
Codewise looks good. so lets merge it.

**Pull request merged**